tasks/main.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: Calculate facts
+  set_fact:
+    build_base: "{{ (mumble_web.build_base | default(mumble_web_build_base)) | regex_replace('\\/$', '') }}"
+    deploy_path: "{{ (mumble_web.deploy_path | default(mumble_web_deploy_path)) | regex_replace('\\/$', '') }}"
+    version: "{{ mumble_web.version }}"
+    checksum: "{{ mumble_web.checksum }}"
+    reinstall: "{{ mumble_web.reinstall | default(false)}}"
+    config: "{{ mumble_web.config | default({}) }}"
+
+- name: Calculate facts for installation
+  set_fact:
+    archive_file: "{{ global_cache_dir }}/mumble-web-{{ version }}.tar.gz"
+    build_dir: "{{ build_base }}/mumble-web-{{ version }}"
+    deploy_base: "{{ deploy_path | dirname }}"
+
+- name: Create deployment parent directory
+  file:
+    path: "{{ deploy_base }}"
+    state: directory
+
+- name: Calculate facts for checking if install is neccesarry
+  set_fact:
+    deployed_version_file: "{{ deploy_base }}/.{{ deploy_path | basename }}_ansible_deployed_version"
+
+- name: Make sure deployed version file exists
+  file:
+    path: "{{deployed_version_file}}"
+    state: touch
+
+- name: Read deployed version file
+  shell: "cat {{deployed_version_file}}"
```

Comment:
You can use [lookups](https://docs.ansible.com/ansible/latest/user_guide/playbooks_lookups.html#id3) for this. Also, I think the `file` module supports this.

Reply:
I was actually trying a lot of things and this is the only one I got to work in the end. 
The [file lookup](https://docs.ansible.com/ansible/latest/plugins/lookup/file.html) only works on the controller’s file system 
and if download it first there were some problems if this file was not present on the remote host and so on.

I'd be happy to change this, if you have a working solution.